### PR TITLE
Default image

### LINKED ISSUE #565: User avatar disappears if the original image is deleted from the phone

### LINKED ISSUE #568: Use generated avatar image as default

### PR DESCRIPTION
**Pull request by vujevits**
fixes #568 
fixes #565 
Use identicon.js for generating default user avatars.

### CODE REVIEW

File: src/ui/screens/onboarding/ProfileScreen.tsx [hunk 1]
```diff
@@ -40,8 +47,18 @@ type Props = DispatchProps & StateProps;
...
     const isFormFilled = props.author.name !== '' && props.author.name !== defaultAuthor.name;
+    const [ identity, setIdentity ] = React.useState<PrivateIdentity | undefined>(undefined);
+    React.useEffect(() => {
+        Swarm.generateSecureIdentity(generateSecureRandom).then(value => {
+            setIdentity(value);
+            getFallbackUserImage(value.publicKey).then(image => {
+                props.onUpdatePicture(image);
+            });
+        });
+    }, []);
```
Comment (vujevits) [inline src/ui/screens/onboarding/ProfileScreen.tsx]: trying out the hooks API. One downside is that useEffect does not accept an `async` function.

File: src/defaultUserImage.ts [hunk 2]
```diff
@@ -28,3 +33,29 @@ export const getDefaultUserImage = async (): Promise<ImageData> => {
...
+        }
+        return `${FILE_PROTOCOL}${path}`;
+    } else if (image.localPath != null && isString(image.localPath)) {
+        return await copyToAppDataDir(getAbsolutePathFromLocalPath(image.localPath), filename);
+    } else {
+        throw Error(`copyImageToApp, unsupported image ${image}`);
+    }
+};
+
+export const createUserImage = (hash = generateUnsecureRandomHexString(15)): ImageData => {
```
Comment (agazso) [inline src/defaultUserImage.ts]: Why `15`? :)

**Comment by agazso** [comment]
Awesome 👍

File: src/defaultUserImage.ts (hunk 2 above)
Comment (vujevits) [reply src/defaultUserImage.ts]: that was the minimum for `Identicon.js` 😄

File: src/models/ReactNativeModelHelper.ts [hunk 3]
```diff
@@ -30,6 +29,9 @@ export class ReactNativeModelHelper implements ModelHelper {
         if (image.uri != null) {
             return getSwarmGatewayUrl(image.uri, this.gatewayAddress);
         }
+        if (image.data != null) {
+            return `data:image/gif;base64,${image.data}`;
```
Comment (agazso) [inline src/models/ReactNativeModelHelper.ts]: Why does this return `image/gif` as mime-type? Everywhere else we use `.png`

File: src/defaultUserImage.ts [hunk 4]
```diff
@@ -28,3 +33,29 @@ export const getDefaultUserImage = async (): Promise<ImageData> => {
         throw new Error('unsupported platform: ' + Platform.OS);
     }
 };
+
+export const copyImageToApp = async (image: ImageData, filename: string): Promise<string> => {
+    if (image.data != null) {
+        const path = await getPathForFile(filename);
+        if (!await RNFS.exists(path)) {
+            try {
+                await RNFS.writeFile(path, image.data, 'base64');
```
Comment (agazso) [inline src/defaultUserImage.ts]: This doesn't seem to be good, the data url header should be stripped

Comment (vujevits) [reply src/defaultUserImage.ts]: it is not there

**Pull request merged**